contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,933 @@
+                    amount1Min: 0,
+                    deadline: block.timestamp
+                })
+            );
+
+        (
+            uint256 _amountWethCollected,
+            uint256 _amountOethbCollected
+        ) = positionManager.collect(
+                INonfungiblePositionManager.CollectParams({
+                    tokenId: tokenId,
+                    recipient: address(this),
+                    amount0Max: type(uint128).max, // defaults to all tokens owed
+                    amount1Max: type(uint128).max // defaults to all tokens owed
+                })
+            );
+
+        _updateUnderlyingAssets();
+
+        emit LiquidityRemoved(
+            _liquidityToDecrease,
+            _amountWeth, //removedWethAmount
+            _amountOethb, //removedOethbAmount
+            _amountWethCollected,
+            _amountOethbCollected,
+            underlyingAssets
+        );
+    }
+
+    /**
+     * @dev Perform a swap so that after the swap the ticker has the desired WETH to OETHb token share.
+     */
+    function _swapToDesiredPosition(
+        uint256 _amountToSwap,
+        uint256 _minTokenReceived,
+        bool _swapWeth
+    ) internal {
+        IERC20 _tokenToSwap = IERC20(_swapWeth ? WETH : OETHb);
+        uint256 _balance = _tokenToSwap.balanceOf(address(this));
+
+        if (_balance < _amountToSwap) {
+            // if swapping OETHb
+            if (!_swapWeth) {
+                uint256 mintForSwap = _amountToSwap - _balance;
+                IVault(vaultAddress).mintForStrategy(mintForSwap);
+            } else {
+                revert NotEnoughWethForSwap(_balance, _amountToSwap);
+            }
+        }
+
+        // emit an event so it is easier to find correct values off-chain
+        (
+            uint256 _wethPositionBalance,
+            uint256 _oethbPositionBalance
+        ) = getPositionPrincipal();
+        emit PrincipalPositionBeforeSwap(
+            _wethPositionBalance,
+            _oethbPositionBalance
+        );
+
+        // Swap it
+        swapRouter.exactInputSingle(
+            // sqrtPriceLimitX96 is just a rough sanity check that we are within 0 -> 1 tick
+            // a more fine check is performed in _checkForExpectedPoolPrice
+            // TBD(!): this needs further work if we want to generalize this approach
+            ISwapRouter.ExactInputSingleParams({
+                tokenIn: address(_tokenToSwap),
+                tokenOut: _swapWeth ? OETHb : WETH,
+                tickSpacing: tickSpacing, // set to 1
+                recipient: address(this),
+                deadline: block.timestamp,
+                amountIn: _amountToSwap,
+                amountOutMinimum: _minTokenReceived, // slippage check
+                sqrtPriceLimitX96: _swapWeth
+                    ? sqrtRatioX96Tick0
+                    : sqrtRatioX96Tick1
+            })
+        );
+
+        // emit an event so it is easier to find correct values off-chain
```

Comment:
🟠Given that we are withdrawing our entire principle before making a swap, won't this event always emit zeros?

Reply:
Good point. Now that we remove 100% of the liquidity emitting this event makes no sense anymore. 
https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/c7d95fefb94625123239a45efdd9981548a35948

Reply:
🟠 This code seems to have come back again. We don't need to emit this event, right?

Reply:
Git blame shows the event change happening [17 days ago in a commit](https://github.com/OriginProtocol/origin-dollar/commit/608d465ce601b261e7e6d93062950c154d2a3095) even though the event [has been removed 5 days ago](https://github.com/OriginProtocol/origin-dollar/commit/c7d95fefb94625123239a45efdd9981548a35948). Confusing

Reply:
removed again: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/38a6af9d7ed9dcabd44a1cba39d9610f2edfb55d